### DISCUSSION TITLE
Role permissions for `argo retry`?

### DISCUSSION THREAD

**Discussion by brews** [argoproj/argo-workflows#8127] [Q&A] [answered]
I have a user who wants to `argo retry` a workflow with a step that failed for a trivial reason — they might try this via the server UI or the CLI. This user is using a k8s service account token to authenticate. When they attempt `argo retry` they get an error about missing "permission to delete pods". I'm unsure if this is the only additional requirement and I haven't RBAC specs for `retry` documented anywhere obvious (...?).

What exactly are the Role permissions required for `argo retry`?

Thanks!

**Comment by terrytangyuan** [comment, marked as answer]
How did you install Argo Workflows? If you used what's in this directory, you should have `pods delete` permission already: https://github.com/argoproj/argo-workflows/tree/master/manifests

**Reply by brews** [reply]
Uh oh, is this something the controller should be handling? I'm happy to file this as an issue instead. Searching around it looks like [one other person might have hit the same issue](https://stackoverflow.com/questions/68582187/what-rbac-permissions-needed-for-argo-retry).

We got the error on a system deploying from `https://github.com/argoproj/argo-workflows/releases/download/v3.2.9/install.yaml` with ArgoCD. It's from [this kustomization](https://github.com/ClimateImpactLab/downscaleCMIP6/tree/master/infrastructure/kubernetes-gcp/argo) for a small research project.

(And thanks for your help!)

**Reply by brews** [reply]
The exact error when this fails in the server UI is
```
rpc error: code = Unknown desc = pods "testworkflow-dev-kbsfv-1115637809" is forbidden: User "system:serviceaccount:default:testsa" cannot delete resource "pods" in API group "" in the namespace "default"
```

**Reply by terrytangyuan** [reply]
Check if your testsa service account has such permission in default namespace.

**Reply by brews** [reply]
Yup. That does the trick, @terrytangyuan. Guess it throws me off because this is the first time I e had to give pod permissions for the "user" SA.
